owncloudApp/src/main/java/com/owncloud/android/shares/ui/fragment/ShareFileFragment.kt

```diff
@@ -425,76 +383,40 @@ class ShareFileFragment : Fragment(), ShareUserListAdapter.ShareUserAdapterListe
-                    }
-                    else -> {
-                        Log.d(TAG, "Unknown status when loading capabilities")
-                    }
-                }
-            }
-        )
-    }
+    fun updateCapabilities(capabilities: OCCapability?) {
+        this.capabilities = capabilities
 
-    private fun loadPublicShares() {
+        updatePublicLinkButton()
+
+        // Update view depending on updated capabilities
         if (isPublicShareDisabled) {
-            hidePublicShare()
+            shareViaLinkSection.visibility = View.GONE
         } else {
-            ocShareViewModel.getSharesForFile().observe(
-                this,
-                Observer { resource ->
-                    when (resource?.status) {
-                        Status.SUCCESS -> {
-                            publicLinks = resource.data as ArrayList<OCShare>
-                            updateListOfPublicLinks()
-                            (activity as BaseActivity).dismissLoadingDialog()
-                        }
-                        Status.ERROR -> {
-                            val errorMessage = ErrorMessageAdapter.getResultMessage(
-                                resource.code,
-                                resource.exception,
-                                OperationType.GET_SHARES,
-                                resources
-                            )
-                            view?.let { Snackbar.make(it, errorMessage, Snackbar.LENGTH_SHORT).show() }
-                            publicLinks = resource.data as ArrayList<OCShare>
-                            updateListOfPublicLinks()
-                            (activity as BaseActivity).dismissLoadingDialog()
-                        }
-                        Status.LOADING -> {
-                            (activity as BaseActivity).showLoadingDialog(R.string.common_loading)
-                            publicLinks = resource.data as ArrayList<OCShare>
-                            updateListOfPublicLinks()
-                        }
-                        else -> {
-                            Log.d(TAG, "Unknown status when loading shares")
-                        }
-                    }
-                }
-            )
+            shareViaLinkSection.visibility = View.VISIBLE
+        }
+    }
+
+    fun updatePublicShares(publicShares: ArrayList<OCShare>) {
+        publicLinks = publicShares
+        updatePublicLinkButton()
+        updateListOfPublicLinks()
+    }
+
+    /**
+     * Show or hide button for adding a new public share depending on the capabilities and the server version
+     */
+    private fun updatePublicLinkButton() {
+        // Since capabilities and publicLinks are loaded asynchronously, let's check whether they both exist
+        if (capabilities == null || publicLinks == null) {
+            return
+        }
+
+        if (!enableMultiplePublicSharing()) {
+            if (publicLinks?.size == 0) {
```

Comment:
why don't you use isNullOrEmpty() ?

Reply:
and the else block wouldn't have condition

Reply:
Done